api-controllers/ServiceFabrikApiController.js

```diff
@@ -361,32 +385,37 @@ class ServiceFabrikApiController extends FabrikBaseController {
     const backupStartedAt = new Date();
     return eventmesh
       .apiServerClient.getLastOperation({
-        resourceId: req.params.instance_id,
+        resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT,
+        resourceType: CONST.APISERVER.RESOURCE_TYPES.DIRECTOR,
         operationName: CONST.OPERATION_TYPE.BACKUP,
         operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+        resourceId: req.params.instance_id
       })
       .then(backupGuid => {
         return eventmesh
-          .apiServerClient.getOperationState({
-            operationName: CONST.OPERATION_TYPE.BACKUP,
-            operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
-            operationId: backupGuid,
+          .apiServerClient.getResource({
+            resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.BACKUP,
+            resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+            resourceId: backupGuid
           })
-          .then(state => {
+          .then(response => {
+            const state = response.status.state;
             // abort only if the state is in progress
             if (state === CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS) {
-              return eventmesh.apiServerClient.updateOperationState({
-                operationName: CONST.OPERATION_TYPE.BACKUP,
-                operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
-                operationId: backupGuid,
-                stateValue: CONST.OPERATION.ABORT
+              return eventmesh.apiServerClient.updateResource({
+                resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.BACKUP,
+                resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+                resourceId: backupGuid,
+                status: {
+                  'state': CONST.OPERATION.ABORT
+                }
               });
             } else {
               logger.info(`Skipping abort for ${backupGuid} as state is : ${state}`);
             }
           })
           .then(() => eventmesh.apiServerClient.getResourceOperationStatus({
-            operationId: backupGuid,
+            resourceId: backupGuid,
             start_state: CONST.OPERATION.ABORT,
             started_at: backupStartedAt
```

Comment:
`also variable is incorrect in abort.` Did not understand this.
And probably it was done because true abort time is start of this method. Can change it if necessary.

Reply:
>also variable is incorrect in abort.

abort start time is set via the variable `backupStartedAt`, hence my comment above.